From b3364b9df48d2c2e56067c14c5d4319f9c5f2e31 Mon Sep 17 00:00:00 2001
From: Beniamin Sandu <beniaminsandu@gmail.com>
Date: Mon, 20 Jul 2020 13:16:29 +0300
Subject: [PATCH 76/78] perf/x86/msr: Add Tremont support

Backport of 0aa0e0d6b34b89649e6b5882a7e025a0eb9bd832 upstream.

Tremont is Intel's successor to Goldmont Plus. SMI_COUNT MSR is also
supported.

Signed-off-by: Kan Liang <kan.liang@linux.intel.com>
Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Signed-off-by: Ingo Molnar <mingo@kernel.org>
Reviewed-by: Andi Kleen <ak@linux.intel.com>
Link: https://lkml.kernel.org/r/1580236279-35492-3-git-send-email-kan.liang@linux.intel.com
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Beniamin Sandu <beniamin.sandu@windriver.com>
---
 arch/x86/events/msr.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/events/msr.c b/arch/x86/events/msr.c
index ace6c1e752fb..c1dd9c708da7 100644
--- a/arch/x86/events/msr.c
+++ b/arch/x86/events/msr.c
@@ -77,6 +77,8 @@ static bool test_intel(int idx)
 	case INTEL_FAM6_ATOM_GOLDMONT_X:
 
 	case INTEL_FAM6_ATOM_GOLDMONT_PLUS:
+	case INTEL_FAM6_ATOM_TREMONT_X:
+	case INTEL_FAM6_ATOM_TREMONT:
 
 	case INTEL_FAM6_XEON_PHI_KNL:
 	case INTEL_FAM6_XEON_PHI_KNM:
-- 
2.17.1

